docs/reference/eql/functions.asciidoc

```diff
@@ -276,4 +277,82 @@ function returns the remaining string.
 Positions are zero-indexed. Negative offsets are supported.
 
 *Returns:* string
+====
+
+[discrete]
+[[eql-fn-wildcard]]
+=== `wildcard`
+Returns `true` if a source string matches one or more provided wildcard
+expressions. Matching is case sensitive.
+
+[%collapsible]
+====
+*Example*
+[source,eql]
+----
+// The two following expressions are equivalent.
+process.command_line == "*start*" or process.command_line == "*config*"
+wildcard(process.command_line, "*start*", "*config*")
+
+// process.command_line = "start explorer.exe"
+wildcard(process.command_line, "*start*")                // returns true
+wildcard(process.command_line, "*start*", "*config*")    // returns true
+wildcard(process.command_line, "*config*")               // returns false
+wildcard(process.command_line, "*config*", "*create*")   // returns false
+
+// process.command_line = [ "start explorer.exe", "create regsvr32.exe" ]
+wildcard(process.command_line, "*create*")              // returns true
+wildcard(process.command_line, "*create*", "*config*")  // returns true
+wildcard(process.command_line, "*start*")               // returns false
+wildcard(process.command_line, "*start*", "*create*")   // returns true
+wildcard(process.command_line, "*start*", "*config*")   // returns false
+
+// case sensitive matching
+wildcard("start explorer.exe", "*start*")                // returns true
+wildcard("start explorer.exe", "*Start*")                // returns false
+
+// null handling
+wildcard(null, "*start*")                                // returns false
+wildcard(null, null)                                     // returns 500 error
+wildcard("start explorer.exe", null)                     // returns 500 error
+wildcard("start explorer.exe", null, "*start*")          // returns 500 error
+wildcard("", null)                                       // returns 500 error
+wildcard("", "")                                         // returns true
+----
+
+*Syntax*
+
+[source,txt]
+----
+wildcard(<source>, <wildcard_exp>[, ...])
+----
+
+*Parameters*
+
+`<source>`::
++
+--
+(Required, string)
+Source string.
+
+If using a field as the argument, this parameter only supports the following
+field datatypes:
+
+* <<keyword,`keyword`>>
+* <<constant-keyword,`constant_keyword`>>
+* <<text,`text`>> field with a <<keyword,`keyword`>> or
+  <<constant-keyword,`constant_keyword`>> sub-field
+
+Fields containing <<array,array values>> use the last array item only.
```

Comment:
> Fields containing <<array,array values>> use the last array item only.

This behavior seemed a little odd to me, but it's not consistent across functions. `endsWith` and `startsWith` only use the _first_ array item. `length` doesn't support arrays at all.

Reply:
I don't know enough about arrays in ES to speak for this. But under the hood, `wildcard` gets converted to LIKE. This array behavior will be consistent with SQL LIKE.

Reply:
@astefan @costin Would one of you mind verifying this statement?

I've testing using arrays of varying length. It seems to reliably use the last array item.

Reply:
@jrodewig @rw-access  I don't think I understand the array argument... `wildcard` gets converted to `value LIKE array_value1 OR value LIKE array_value2 OR value LIKE array_value3.....`.
With array values in ES, SQL is at the mercy of Lucene. Whatever "first" value Elasticsearch gives us, that's the one we return.

In SQL documentation we say about `field_multi_value_leniency` parameter: `Throw an exception when encountering multiple values for a field (default) or be lenient and return the **first value from the list (without any guarantees of what that will be - typically the first in natural ascending order)**.` So "first"/"last" has loose meanings in this context. And Elasticsearch itself doesn't offer any guarantees of the ordering of values in a potential array of values for a certain field, because ES doesn't care too much about the ordering. It's typically ascending order, but it's not guaranteed to be like this always and for every field/values.

Reply:
Thanks for the explanation @astefan.

In my opinion, it seems like documenting a limitation similar to the [SQL one][0] makes sense. Generally, we should discourage users from using array values with EQL and explain the perils.

I raised https://github.com/elastic/elasticsearch/issues/54970 as a discuss issue to get consensus before moving forward. If we decide to document the limitation, I'll remove the array value references from the `wildcard` function docs.

[0]: https://www.elastic.co/guide/en/elasticsearch/reference/current/sql-limitations.html#_array_type_of_fields